jx_sqlite/expressions.py

```diff
@@ -133,18 +134,32 @@ def to_sql(self, schema, not_null=False, boolean=False):
     acc = []
     if len(lhs) != len(rhs):
         Log.error("lhs and rhs have different dimensionality!?")
-    for l, r in zip(lhs, rhs):
-        for t in "bsnj":
-            if l.sql[t] == None:
-                if r.sql[t] == None:
-                    pass
-                else:
-                    acc.append("(" + r.sql[t] + ") IS NULL")
-            else:
-                if r.sql[t] == None:
-                    acc.append("(" + l.sql[t] + ") IS NULL")
+    
+    if boolean:
+        for l, r in zip(lhs, rhs):
+            for t in "bsnj":
+                if l.sql[t] == None:
+                    if r.sql[t] == None:
+                        pass
+                    else:
+                        acc.append("(" + r.sql[t] + ") IS NULL")
                 else:
-                    acc.append("((" + l.sql[t] + ") = (" + r.sql[t] + ") OR ((" + l.sql[t] + ") IS NULL AND (" + r.sql[t] + ") IS NULL))")
+                    if r.sql[t] == None:
+                        acc.append("(" + l.sql[t] + ") IS NULL")
+                    else:
+                        if is_column_name(dequote(l.sql[t])) and is_column_name(dequote(r.sql[t])):
```

Comment:
Let me look at this problem.  You have tried a few times, and each time you are "patching"; detecting a symptom, and adding code to recover from it. In many applications, especially human-machine applications, this strategy is a tolerable short term solution because they do not need to deal with complexity, but this library is performing a complex data mapping, and we need the code clean to keep it understandable. Solutions like you have proposed are pure code debt: A couple more patches like this will quickly l compound the logic flow to the point where no human will understand.

Take you time to figure out why this code is wrong; the solution is probably nowhere near this code. Do not rush.  

Like I said at the beginning:  I will look a this and get back to you on how it should b solved.